### PR TITLE
[merged] syscontainers: environment variable detection

### PR DESCRIPTION
**Pull request by yuqi-zhang**
Add functionality to detect environment variables in template files.
The variables that can be set by the user through --set is displayed
on 'atomic images info'. The variables with overridable default
values will either have their default value displayed, or shown as
{SET_BY_OS} if atomic sets them. Variables without default value
must be set by the user and is listed separately.

Also add check to install of system containers to error out if a
value is required but not set.

Signed-off-by: Yu Qi Zhang <jerzhang@redhat.com>

### CODE REVIEW

**Comment by rhatdan** [comment]
@giuseppe PTAL

File: Atomic/syscontainers.py [hunk 1]
```diff
@@ -306,6 +309,23 @@ def _do_checkout(self, repo, name, img, upgrade, values, destination, unitfileou
         if rev is None:
             raise ValueError("Image %s not found" % img)
 
+        if not upgrade and not extract_only:
```
Comment (giuseppe) [inline Atomic/syscontainers.py]: we can use `--set` on `update --container` as well.  Can we drop the `not upgrade` condition here?

Comment (yuqi-zhang) [inline Atomic/syscontainers.py]: As pointed out the check is redundant, and I have removed it.

**Comment by giuseppe** [comment]
@rh-atomic-bot r+ 981d644

**Comment by rhatdan** [comment]
@rh-atomic-bot retry

**Comment by rhatdan** [comment]
@rh-atomic-bot r+

**Comment by giuseppe** [comment]
@rh-atomic-bot r+

**Comment by rhatdan** [comment]
@jlebon Is something wrong with the BOT?

**Comment by jlebon** [comment]
@rh-atomic-bot retry

**Comment by jlebon** [comment]
It's been feeling a little under the weather recently.

**Pull request closed without merging**